models/selective_triggers.go

```diff
@@ -0,0 +1,381 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+type Triggers struct {
+	PushTriggers        []PushGitEventTriggerItem        `json:"push,omitempty" yaml:"push,omitempty"`
+	PullRequestTriggers []PullRequestGitEventTriggerItem `json:"pull_request,omitempty" yaml:"pull_request,omitempty"`
+	TagTriggers         []TagGitEventTriggerItem         `json:"tag,omitempty" yaml:"tag,omitempty"`
+}
+
+type PushGitEventTriggerItem struct {
+	Enabled       *bool `json:"enabled,omitempty" yaml:"enabled,omitempty"`
+	Branch        any   `json:"branch,omitempty" yaml:"branch,omitempty"`
+	CommitMessage any   `json:"commit_message,omitempty" yaml:"commit_message,omitempty"`
+	ChangedFiles  any   `json:"changed_files,omitempty" yaml:"changed_files,omitempty"`
+}
+
+type PullRequestGitEventTriggerItem struct {
+	Enabled       *bool `json:"enabled,omitempty" yaml:"enabled,omitempty"`
+	DraftEnabled  *bool `json:"draft_enabled,omitempty" yaml:"draft_enabled,omitempty"`
+	SourceBranch  any   `json:"source_branch,omitempty" yaml:"source_branch,omitempty"`
+	TargetBranch  any   `json:"target_branch,omitempty" yaml:"target_branch,omitempty"`
+	Label         any   `json:"label,omitempty" yaml:"label,omitempty"`
+	Comment       any   `json:"comment,omitempty" yaml:"comment,omitempty"`
+	CommitMessage any   `json:"commit_message,omitempty" yaml:"commit_message,omitempty"`
+	ChangedFiles  any   `json:"changed_files,omitempty" yaml:"changed_files,omitempty"`
+}
+
+type TagGitEventTriggerItem struct {
+	Enabled *bool `json:"enabled,omitempty" yaml:"enabled,omitempty"`
+	Name    any   `json:"name,omitempty" yaml:"name,omitempty"`
+}
+
+func (pushItem PushGitEventTriggerItem) toString() string {
```

Comment:
nit: why not `Stringer.String()`? https://pkg.go.dev/fmt#Stringer

Reply:
Because these `toString` functions are helper functions for finding duplicated items, the Enabled property is omitted when comparing two items.